persistence/src/main/java/edu/softserve/zoo/persistence/provider/JdbcTemplatePersistenceProvider.java

```diff
@@ -0,0 +1,25 @@
+package edu.softserve.zoo.persistence.provider;
+
+import edu.softserve.zoo.persistence.exception.PersistenceProviderException;
+import edu.softserve.zoo.persistence.specification.hibernate.SqlGetMapSpecification;
+
+import java.util.Map;
+
+/**
+ * JdbcTemplate based persistence provider for native SQL queries
+ *
+ * @author Serhii Alekseichenko
+ */
+public interface JdbcTemplatePersistenceProvider {
+
+    /**
+     * Get parametrized map from the persistent storage according to given Specification.
+     * In most keys key is an Id and value is a result of an aggregation function
+     *
+     * @param specification defines restrictions for performed search.
+     * @return map of Key-Values pairs
+     * @throws PersistenceProviderException if the received specification is null
+     */
+    <K extends Number, V extends Number> Map<K, V> getMap(SqlGetMapSpecification<K, V> specification);
```

Comment:
Maybe the main reason - is that I consider this functional as a tool to return the ratio of the result of the aggregate function to id. Can you give me some advice on this?